Marsiple/App/Todos/TodosViewController.swift

```diff
@@ -115,17 +101,52 @@ private extension TodosViewController {
         return [uncompleted, completed]
     }
     
-    func reloadData() {
-        let newlyCompleted = todos[TodosIndex.uncompleted.rawValue].filter { $0.completed }
-        if !newlyCompleted.isEmpty {
-            todos[TodosIndex.uncompleted.rawValue] = todos[TodosIndex.uncompleted.rawValue].filter { $0.completed == false }
-            todos[TodosIndex.completed.rawValue] += newlyCompleted
+    func didSelectCheckBox(atIndexPath indexPath: IndexPath) -> (() -> Void)? {
+        let didSelectCheckBox = { [weak self] in
+            // TODO: Save changes on the server
+            guard let strongSelf = self
+                else { return }
+            
+            let finishIndexPath = IndexPath(row: indexPath.section * strongSelf.todos[1 - indexPath.section].count, section: 1 - indexPath.section)
+            var reloadTableView: (Bool) -> Void
```

Comment:
Sorry, I'm not sure I follow - why is the `reloadTableView` closure different depending on the `section`? In both cases, we should just `move` and `reload`, only the `input index paths` change.